ndonnx/_core/_stringimpl.py

```diff
@@ -0,0 +1,50 @@
+# Copyright (c) QuantCo 2023-2024
+# SPDX-License-Identifier: BSD-3-Clause
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import numpy as np
+
+import ndonnx as ndx
+import ndonnx._data_types as dtypes
+import ndonnx._opset_extensions as opx
+
+from ._default import OperationsBlock
+from ._utils import binary_op
+
+if TYPE_CHECKING:
+    from ndonnx import Array
+
+
+class StringOperationsImpl(OperationsBlock):
+    def add(self, x, y) -> Array:
+        return binary_op(x, y, opx.string_concat)
+
+    def equal(self, x, y) -> Array:
+        return binary_op(x, y, opx.equal)
+
+    def can_cast(self, from_, to) -> bool:
```

Comment:
Isn't this a reasonable default implementation?

Reply:
I'm a bit confused as to what you're referring to with "this".

Reply:
Yes, I'm referring to this implementation of `can_cast` which appears to be copied three times.

Reply:
It's there 3 times since we have 3 disjoint data type implementations. It's not worth adding a superclass where it is implemented once just for this.

Reply:
It's there 3 times since we have 3 disjoint data type implementations. It's not worth adding a superclass where it is implemented once just for this. I don't understand what you mean by "default" implementation?

Reply:
I mean that instead of copying it three times we may want to put it [here](https://github.com/Quantco/ndonnx/blob/118ecdc19ba9e20f003ed4ca93412be9da22b679/ndonnx/_core/_interface.py#L411). It is even a reasonable default for a custom type, isn't it?

Reply:
Potentially but let's keep `_interface` actually just an interface rather than blessing a few functions with default implementations (it used to be like this and it's unclear what you get for free).